src/service/base/NativeMetaDb.cpp

```diff
@@ -132,37 +131,32 @@ bool NativeMetaDb::backup() noexcept {
   if (dirty_) {
     dirty_ = false;
     auto uriInfo = nebula::storage::parse(remote_);
-    if (uriInfo.schema == "s3") {
-      // sync all remote data into a local folder
-      auto fs = nebula::storage::makeFS("s3", uriInfo.host);
-
-      // generate a new unique snapshot
-      auto version = std::to_string(nebula::common::Evidence::unix_timestamp());
-      auto snapshot = fmt::format("{0}/{1}", uriInfo.path, version);
-
-      // failed to upload this snapshot
-      if (!fs->sync(backup_, snapshot)) {
-        return false;
-      }
-
-      // write this version into version file
-      auto versionFile = fmt::format("{0}/version", backup_);
-      auto remoteVersion = fmt::format("{0}/version", uriInfo.path);
-      std::ofstream ofs;
-      ofs.open(versionFile, std::ofstream::out);
-      ofs << version;
-      ofs.close();
-
-      // upload this file into S3
-      auto versionUpdated = fs->copy(versionFile, remoteVersion);
-
-      // clean backup folder for next sync
-      resetBackup();
-      return versionUpdated;
+    // sync all remote data into a local folder
+    auto fs = nebula::storage::makeFS(uriInfo.schema, uriInfo.host);
+
+    // generate a new unique snapshot
+    auto version = std::to_string(nebula::common::Evidence::unix_timestamp());
+    auto snapshot = fmt::format("{0}/{1}", uriInfo.path, version);
+
+    // failed to upload this snapshot
+    if (!fs->sync(backup_, snapshot)) {
+      return false;
     }
 
-    // not supported location
-    LOG(WARNING) << "Unsupported backup location: " << remote_;
+    // write this version into version file
+    auto versionFile = fmt::format("{0}/version", backup_);
```

Comment:
Shall we make version as constant since it is used many times.
inline constexpr std::string_view kVersion = "version";